tests/auh/managers/test_base_auth_manager.py

```diff
@@ -0,0 +1,35 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import pytest
+
+from airflow.auth.managers.base_auth_manager import BaseAuthManager
+
+
+@pytest.fixture
+def auth_manager():
+    class EmptyAuthManager(BaseAuthManager):
+        def get_user_name(self) -> str:
+            raise NotImplementedError()
+
+    return EmptyAuthManager()
+
+
+class TestBaseAuthManager:
+    def test_get_security_manager_override_class_return_class(self, auth_manager):
+        assert type(auth_manager.get_security_manager_override_class()) is type
```

Comment:
Why do we need this test? If the goal is to make sure the function returns a type, the check should be `issubclass(..., object)`. `is type` would check the return value can only be a `type` _exactly_, which would break when a metaclass is involved (metaclasses in Python are implemented by subclassing `type`, and that would fail the `is` check).

Reply:
Good call. I think I fixed it it another PR but I should have done it here. Update to `assert auth_manager.get_security_manager_override_class() is object` which basically check that the function return `object`. Let me know if this is good for you

Reply:
Oh so we do want to check the return value is _exactly_ `object`? OK then.